File: rules-interpreter/src/main/java/com/github/iamrenny/ruleflow/evaluators/ValueContextEvaluator.java
package com.github.iamrenny.ruleflow.evaluators;

import com.github.iamrenny.ruleflow.RuleFlowLanguageParser.ValueContext;
import com.github.iamrenny.ruleflow.visitors.Visitor;

import java.math.BigDecimal;
import java.time.LocalDateTime;


public class ValueContextEvaluator implements ContextEvaluator<ValueContext> {

    @Override
    public Object evaluate(ValueContext ctx, Visitor visitor) {
        if (ctx.validValue().string != null) {
            return (ctx.validValue().string.getText()).replace("'", "");
        } else if (ctx.validValue().number != null) {
            return Double.valueOf(ctx.validValue().number.getText());
        } else if (ctx.validValue().nullValue != null) {
            return ctx.validValue().nullValue;
        } else if (ctx.validValue().currentDate != null) {
            return LocalDateTime.now();
        } else if (ctx.validValue().booleanLiteral != null) {
            return Boolean.parseBoolean(ctx.validValue().booleanLiteral.getText());
        } else {
            throw new IllegalArgumentException("Value not supported: " + ctx.getText());
        }
    }
}

File: rules-interpreter/src/main/kotlin/com/rappi/analang/evaluators/ValueContextEvaluator.kt
package com.github.iamrenny.rulesflow.evaluators

import com.github.iamrenny.rulesflow.removeSingleQuote
import com.github.iamrenny.rulesflow.visitors.Visitor
import com.rappi.fraud.analang.ANAParser.ValueContext
import java.time.LocalDateTime

class ValueContextEvaluator : ContextEvaluator<ValueContext> {

    override fun evaluate(ctx: ValueContext, visitor: Visitor): Any {
        return when {
            ctx.validValue().string != null -> ctx.validValue().string.text.removeSingleQuote()
            ctx.validValue().number != null -> ctx.validValue().number.text.toBigDecimal()
            ctx.validValue().nullValue != null -> ctx.validValue().nullValue
            ctx.validValue().currentDate != null -> LocalDateTime.now()
            ctx.validValue().booleanLiteral != null -> ctx.validValue().booleanLiteral.text.toBoolean()
            else -> throw IllegalArgumentException("Value not supported: ${ctx.text}")
        }
    }
}